API/pom.xml

```diff
@@ -15,8 +15,8 @@
 	<description>URLFrontier Java API</description>
 
 	<properties>
-		<grpc.version>1.53.0</grpc.version>
```

Comment:
It would be better to deal with dependency updates in a separate PR and have in the current one only the changes required for the functionality you want to add.  Unless of course this is required to have the tests you added

Reply:
Agreed, will revert this change

Reply:
The same applies to 	`<artifactId>protobuf-java-util</artifactId>`
+ the version prob needs to be kept in sync with that of grpc